### PR TITLE
astropy#14818 - Adress ruff rule ISC003 violations

### PR DESCRIPTION
**Pull request by arthurxvtv**
### Description

Currently there are multiple files that violate ruff rule ISC003. Rule ISC003 checks for string literals that are explicitly concatenated (using the + operator). 

This pull request is to address Issue #14818

### CODE REVIEW

File: astropy/io/ascii/fixedwidth.py
Comment (eerovaher) [inline astropy/io/ascii/fixedwidth.py:74]: This change looks like it might conflict with [ISC001](https://beta.ruff.rs/docs/rules/single-line-implicit-string-concatenation/).

**Review by eerovaher — COMMENTED** [review]
I noticed that there is at least one change that conflicts with Ruff rule [ISC001](https://beta.ruff.rs/docs/rules/single-line-implicit-string-concatenation/). That rule is still in the ignore list, but there already is a pull request that addresses that rule: #14911. Perhaps it is simplest to wait here until #14911 is merged and then to rebase this pull request to ensure that both rules are enforced.

File: astropy/wcs/wcs.py
Comment (mcara) [inline astropy/wcs/wcs.py:3122]: I do not like proposed change here. Much less clear than `+`. Suggestion:

```suggestion
        sfmt = " : " + ""..join([f"{{{i}!r}} " for i in range(self.naxis)])
```

Comment (nstarman) [reply astropy/wcs/wcs.py:3122]: @mcara I think there's an extraneous "."

Comment (nstarman) [reply astropy/wcs/wcs.py:3122]:
```suggestion
        sfmt = " : " + "".join([f"{{{i}!r}} " for i in range(self.naxis)])
```

Comment (mcara) [reply astropy/wcs/wcs.py:3122]: Yes, I made a mistake.

Comment (eerovaher) [reply astropy/wcs/wcs.py:3122]: This is another example of a change that conflicts with Ruff rule [ISC001](https://beta.ruff.rs/docs/rules/single-line-implicit-string-concatenation/), but that rule is being addressed in a separate pull request.

**Comment by eerovaher** [comment]
#14911 is finally merged, so this should now be rebased on current `main`.

File: astropy/io/ascii/fixedwidth.py
Comment (taldcroft) [reply astropy/io/ascii/fixedwidth.py:74]: Agreed on the conflict with implicit string concatenation. But more importantly here the original is really preferred due to the high density of special characters.

**Review by taldcroft — COMMENTED** [review]
Looks good apart from one comment.

File: astropy/io/ascii/fixedwidth.py
Comment (nstarman) [reply astropy/io/ascii/fixedwidth.py:74]: would 
```suggestion
    set_of_position_line_characters = set(r"""`~!#$%^&*-_+=\|":'""")
```
work?

File: astropy/wcs/wcs.py
Comment (nstarman) [reply astropy/wcs/wcs.py:3122]: Whichever is merged first will have to rebase and deal with the change. LGTM.

**Comment by nstarman** [comment]
@arthurxvtv, If you rebase onto main and address the one comment then this PR can be merged. Thanks for your patience!

File: astropy/io/ascii/fixedwidth.py
Comment (taldcroft) [reply astropy/io/ascii/fixedwidth.py:74]: Good idea, fine by me!

**Comment by nstarman** [comment]
pre-commit.ci autofix

**Comment by nstarman** [comment]
Darn. Not auto fixable. @mcara's comment looks like it will resolve this issue.

**Review by nstarman — APPROVED** [review]
Thanks @arthurxvtv. LGTM! let's squash-merge this in...

**Review by eerovaher — CHANGES_REQUESTED** [review]
I have two remarks. One is best left for a separate pull request, but the other should be addressed here.

File: astropy/io/votable/tree.py
Comment (dhomeier) [reply astropy/io/votable/tree.py:2945]: The supported `href`s would include https btw., perhaps this should be clarified nowadays.

File: astropy/wcs/utils.py
Comment (nstarman) [reply astropy/wcs/utils.py:1047]: A good catch.

Comment (nstarman) [reply astropy/wcs/utils.py:1047]: converted your comment into an issue.

File: astropy/io/votable/tree.py
Comment (nstarman) [reply astropy/io/votable/tree.py:2945]: Or we could just tack on a `# noqa: E501`.

Comment (nstarman) [reply astropy/io/votable/tree.py:2945]: Nope. Looks good.

File: astropy/io/votable/tree.py
Comment (nstarman) [inline astropy/io/votable/tree.py:2944]:
```suggestion
                    (
                        "The vo package only supports remote data through http, "
```

File: astropy/io/votable/tree.py
Comment (nstarman) [inline astropy/io/votable/tree.py:2945]:
```suggestion
                        "ftp or file"
                    ),
```

**Review by nstarman — COMMENTED** [review]
YOLO from the browser.

File: astropy/io/votable/tree.py
Comment (nstarman) [reply astropy/io/votable/tree.py]: @dhomeier made this into an issue, because I found it in more than one location.

File: astropy/io/votable/tree.py
Comment (mcara) [inline astropy/io/votable/tree.py:2947]: I think line continuation would be much better than these extra parenthesis and extra indentation. Why is this needed? I think this will work too:
```suggestion
                    "The vo package only supports remote data through http, "
                    "ftp or file",
```

Comment (eerovaher) [reply astropy/io/votable/tree.py:2947]: How simple would you say it is to figure out the difference between the following code snippets?
```python
vo_raise(
    "The vo package only supports remote data through http, "
    "ftp or file",
    self._config,
    self._pos,
    NotImplementedError,
)
```
```python
vo_raise(
    "The vo package only supports remote data through http, ",
    "ftp or file",
    self._config,
    self._pos,
    NotImplementedError,
)
```

Comment (dhomeier) [reply astropy/io/votable/tree.py:2947]: I get your point about confusion with extra args now, but still don't find the extra parentheses on extra lines terribly readable either. Perhaps just put the entire string on one line (and all other args onto another) and protect them from black, after all?

Or try something completely different:
```python
           vo_prot =  ['http', 'https', 'ftp', 'file']
           if href.split(":")[0] not in vo_prot:
                vo_raise(
                    f"The vo package only supports remote data through {vo_prot}",
```
That would also keep bogus protocols like `httpx://...` or `filename:...` from being passed to urllib.

Comment (mcara) [reply astropy/io/votable/tree.py:2947]: > How simple would you say it is to figure out the difference between the following code snippets?

I actually have no problem figuring out differences right away. And probably any unit test would fail too.

File: astropy/io/votable/tree.py
Comment (mcara) [reply astropy/io/votable/tree.py]: > Or we could just tack on a `# noqa: E501` and make it a single line.

I like this suggestion too. Most modern screens can handle slightly more characters

File: astropy/io/votable/tree.py
Comment (nstarman) [reply astropy/io/votable/tree.py]: Done!

**Review by mcara — APPROVED** [review]
OK for WCS

File: astropy/io/votable/tree.py
Comment (nstarman) [reply astropy/io/votable/tree.py]: ping @dhomeier @eerovaher if you approve this change.

File: astropy/io/votable/tree.py
Comment (dhomeier) [inline astropy/io/votable/tree.py:2943]: This still does not check if it exactly matches the pattern _"vo_prot://..."_, but since that would technically change error handling, it's OK to follow this up in #15194.

**Review by dhomeier — APPROVED** [review]
There is one other instance of the `href` check in tree.py#L3061 which probably could be updated along with the one here, but otherwise LGTM now.

**Review by eerovaher — COMMENTED** [review]
This could be merged after squashing, but there are commits from multiple authors so squashing should be done carefully.

**Comment by nstarman** [comment]
@eerovaher, squashed, keeping newer committer separate.

**Review by eerovaher — APPROVED** [review]

**Pull request merged**
